Update component “dropdown-multi-selects” (#72)
Zywave/booster: src/_areas/design-system/components/dropdown-multi-selects.md

---
title: Dropdown multi-select
subtitle: Type of input field in which a user fills an input box with one or
  many items from a list of acceptable inputs.
api: https://cdn.zywave.com/@zywave/zui-select@next/dist/custom-elements.json
demo: https://cdn.zywave.com/@zywave/zui-select@next/demo/index.html
mainComponentName: zui-select-dropdown
includedElements:
  - zui-select-dropdown
  - zui-option
  - zui-option-group
---
## What is a Dropdown multi-select?

A Dropdown multi-select is a type of input field in which a user fills an input box with one or many items from a list of acceptable inputs.

- - -

## Usage

> Appropriate for lists larger than 2 options that may require a user to select more than 1 option.

Reviewer: ```suggestion
A form control used when there are more than 2 options and may require a user to select more than 1 option.
```

![Dropdown multi-select usage](/images/multi_select_usage.svg)

- - -

## Anatomy

The anatomy of a dropdown multi-select.

![Dropdown multi-select anatomy](/images/multi_select_anatomy.svg)

1. **Input:** The field designated for selecting an individual option from a grouping of related options.

> 2. **Chevron:** The chevron arrow, within the input suggesting that there are items to select within a dropdown.

Reviewer: ```suggestion
2. **Chevron:** The chevron icon indicates there are options available in a dropdown list.
```

3. **Option list:** The grouping of related input options.
4. **Option:** A unique input option.
5. **Option checkbox:** A selectable checkbox beside an option.
6. **Scrollbar**: A bar used to scroll to view more options

<br>

See the [](https://xd.adobe.com/view/ef2f902b-219f-4e41-8bba-2bf079fc5969-ba7c/grid)[Design specs](https://xd.adobe.com/view/630718e2-ca07-4189-961b-2a655245666d-3654/grid) for detailed sizing and spacing information

- - -

## Types

#### Grouping

* Sometimes you'll need to add grouping to a long list of options. Titles can break up options into groups to help users more easily locate them.

![Dropdown multi-select grouping](/images/multiselect_grouping_specs.svg)

<br>

#### Icons

> * Use icons to distinguish between options.

Reviewer: ```suggestion
Use icons to distinguish between option types.
```

> * Often icons are used when you have two different types of options in a list.

Reviewer: Remove this line

Reviewer: The second line

![Dropdown multi-select icons](/images/multiselect_icon_specs.svg)

- - -

## States

#### Input states

![Dropdown multi-select states](/images/multiselect_states.svg)

<br>

#### Option list states

![Dropdown multi-select list states](/images/multiselect_list_states.svg)

- - -

> ## Behavior

Reviewer: Some do/do not examples would be great in this section!

> Sort the list in a logical order. For instance, put the most selected option(s) at the top. In many cases we recommend alphabetical order.

Reviewer: ```suggestion
#### Options list
Sort the list in a logical order. For instance, alphabetical order or put the most selected option(s) at the top.
```

* Dropdrown chevron icon: rotates 180 degrees clockwise when opened, when closed spins back 180 degrees counter clockwise, returning to the original position.
* The dropdown will need to be closed manually after making selections.

#### Scrolling

* If the options can be displayed within the viewport it is ideal to not have a scroll bar. Scroll bars should appear when an options list exceeds the height of the screen but should not be taller than 300px.
* Note options lists will appear below or above select inputs depending on where they sit within the viewport (see Open / close).

#### Typeahead

* Present in all dropdown selects, streamlines experience for users.
* Typeahead is relevant only to options in the list, it will not return group names. Grouping headers are not seen when using typeahead.

[See typeahead](/design-system/patterns/typeahead/)

#### Defaulting a selection

* A dropdown select shouldn't default a selected option. If you're confident users will likely choose an option you may consider defaulting to a selection.

> * Sometimes choosing 'none' as an option is necessary, if none of the options are applicable to the user. None should appear at the top of the list.

Reviewer 2: ```suggestion
* Sometimes choosing "None" as an option is necessary if no options are applicable to the user when a Dropdown multi-select is required. "None" should appear at the top of the list.
```
This is a little odd to point out, but if it's a frequent UX question, may be nice to elaborate. Should "None" only be applicable when the dropdown multi-select is a required field? What if this field isn't required at all? Users should be able to leave it blank without selecting "None".

#### Selection wrapping

> * When multiple options are selected in a multi-select list, selections will populate the input and eventually will have to wrap, in order to fit.

Reviewer 2: ```suggestion
* When options are selected from the Dropdown multi-select option list, selections will populate inside the select box. Multiple selections should wrap within the select box to fit all.
```

![Tags selection wrapping](/images/tag_specs.svg)

#### Removing the selection

> * After a user has chosen an option it turns into a selection within the input. Users can remove a selection by using one of three methods.
> * Selections can be removed by clicking on the X icon within the selection.
> * Backspacing over a selection in the input removes it.
> * By deselecting the checkbox within an option list will also remove the selection.

Reviewer 2: ```suggestion
* After a user has chosen an option it appears as a selection within the select box. Users can remove a selection by using one of three methods:
  * Selections can be removed by clicking on the X icon within the selection.
  * Backspacing over a selection in the select box removes it.
  * Deselecting the checkbox within the option list will also remove the selection.
```
